apps/api/src/domain/models/context/event-sourced-note.view-model.ts

```diff
@@ -0,0 +1,150 @@
+        if (!dto) {
+            return;
+        }
+
+        const { type, id, context } = dto;
+
+        this.type = type;
+
+        this.id = id;
+
+        // @ts-expect-error Deal with the generic type here properly
+        this.context = context;
+    }
+}
+
+class ConnectedResources {
+    to?: EdgeConnectionMemberViewModel;
+    from?: EdgeConnectionMemberViewModel;
+    self?: EdgeConnectionMemberViewModel;
+
+    constructor(dto: DTO<ConnectedResources>) {
+        if (!dto) {
+            return;
+        }
+
+        const { to, from, self } = dto;
+
+        if (to) {
+            this.to = new EdgeConnectionMemberViewModel(to);
+        }
+
+        if (from) {
+            this.from = new EdgeConnectionMemberViewModel(from);
+        }
+
+        if (self) {
+            this.self = new EdgeConnectionMemberViewModel(self);
+        }
+    }
+}
+@CoscradDataExample<EventSourcedNoteViewModel>({
+    example: {
+        type: AggregateType.note,
+        id: buildDummyUuid(5),
+        name: buildMultilingualTextWithSingleItem('breeze'),
+        note: buildMultilingualTextWithSingleItem('this is the note for breeze'),
+        audio: MultilingualAudio.buildEmpty(),
+        tags: [],
+        connectedResources: {
+            self: {
+                context: {
+                    type: EdgeConnectionContextType.general,
+                },
+                type: ResourceType.term,
+                id: buildDummyUuid(6),
+            },
+        },
+        connectionType: EdgeConnectionType.self,
+    },
+})
+export class EventSourcedNoteViewModel {
+    type: CategorizableType = CategorizableType.note;
+
+    @UUID({
+        label: 'id',
+        description: 'system identifier for this resource',
+    })
+    id: AggregateId;
+
+    connectionType: EdgeConnectionType;
+
+    note: MultilingualText;
+
+    connectedResources: ConnectedResources;
+
+    @NestedDataType(TagViewModel, {
+        label: 'tags',
+        description: 'a summary of the tags that have been applied to this resource',
+        isArray: true,
+    })
+    tags: EventSourcedTagViewModel[];
+
+    @NestedDataType(MultilingualText, {
+        label: 'name',
+        // note that we call it `name` not `text` for consistency with other models
+        description: 'name (text) includes the text as well as any translations for this term',
+    })
+    name: MultilingualText;
+
+    @NestedDataType(MultilingualAudio, {
+        label: 'audio',
+        description: 'audio to accompany the note and its translations',
+    })
+    audio: MultilingualAudio;
+
+    constructor({
+        connectionType,
+        note,
+        connectedResources,
+        id,
+        tags,
+        name,
+        audio,
+    }: DTO<EventSourcedNoteViewModel>) {
+        this.id = id;
+
+        this.connectionType = connectionType;
+
+        this.note = new MultilingualText(note);
+
+        this.connectedResources = connectedResources;
```

Comment:
if(connectedResources){
this.connectedResources = new ConnectedResources(connectedResources);
}

Reply:
done